src/node_buffer.cc

```diff
@@ -310,6 +310,70 @@ void Base64Slice(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+// bytesCopied = buffer.mask(mask, target, targetOffset, sourceStart, sourceEnd)
+void Mask(const FunctionCallbackInfo<Value> &args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  uint32_t mask32 = args[0]->Uint32Value();
+  char* mask = (char*)&mask32;
+
+  Local<Object> target = args[1]->ToObject(env->isolate());
+
+  if (!HasInstance(target))
+    return env->ThrowTypeError("second arg should be a Buffer");
+
+  ARGS_THIS(args.This())
+  size_t target_length = target->GetIndexedPropertiesExternalArrayDataLength();
+  char* target_data = static_cast<char*>(
+      target->GetIndexedPropertiesExternalArrayData());
+  size_t target_start;
+  size_t source_start;
+  size_t source_end;
+
+  CHECK_NOT_OOB(ParseArrayIndex(args[2], 0, &target_start));
+  CHECK_NOT_OOB(ParseArrayIndex(args[3], 0, &source_start));
+  CHECK_NOT_OOB(ParseArrayIndex(args[4], obj_length, &source_end));
+
+  // Copy 0 bytes; we're done
+  if (target_start >= target_length || source_start >= source_end)
+    return args.GetReturnValue().Set(0);
+
+  if (source_start > obj_length)
+    return env->ThrowRangeError("out of range index");
+
+  if (source_end - source_start > target_length - target_start)
+    source_end = source_start + target_length - target_start;
+
+  uint32_t to_copy = MIN(MIN(source_end - source_start,
+                             target_length - target_start),
+                             obj_length - source_start);
+
+  obj_data += source_start;
+  target_data += target_start;
+
+  uint32_t* target_data_32 = (uint32_t*)target_data;
+  uint32_t* obj_data_32 = (uint32_t*)obj_data;
```

Comment:
How should I fix this? Is the goal to only have one name in any frame point to a given pointer address? If so, would it be possible to add something like:

```cpp
static inline size_t Mask32(uint32_t* obj_data, uint32_t* target_data, uint32_t mask, size_t len) {
  len = len / 4;
  for (size_t i = 0; i < len; ++i) {
    target_data[i] = obj_data[i] ^ mask;
  }
  return len * 4;
}

// then use it like so:
size_t written = Mask32(
  reinterpret_cast<uint32_t*>(obj_data),
  reinterpret_cast<uint32_t*>(target_data),
  mask.as_dword,
  to_copy
);
target_data += written;
obj_data += written;

```